templates/static/js/example.js

```diff
@@ -3,6 +3,48 @@
     throw Error('VANILLA_VERSION not specified.');
   }
 
+  /**
+   * Mapping of example keys to the regex patterns used to strip them out of an example
+   * @type {{body: RegExp, jinja: RegExp, title: RegExp, head: RegExp}}
+   */
+  const EXAMPLE_CONTENT_PATTERNS = {
+    body: /<body[^>]*>((.|[\n\r])*)<\/body>/im,
+    jinja: /{% block content %}([\s\S]*?){% endblock %}/,
+    title: /<title[^>]*>((.|[\n\r])*)<\/title>/im,
+    head: /<head[^>]*>((.|[\n\r])*)<\/head>/im,
+  };
+
+  /**
+   * Object representing the structure for language option mappings.
+   * @typedef {Object} ExampleLanguageConfig
+   * @property {string} label - Human-readable label.
+   * @property {string} langIdentifier - Prism language identifier.
+   */
+
+  /**
+   * Mapping of example keys to their configurations.
+   * @type {{jinja: ExampleLanguageConfig, css: ExampleLanguageConfig, js: ExampleLanguageConfig, html: ExampleLanguageConfig}}
+   */
+  const EXAMPLE_OPTIONS_CFG = {
+    html: {
+      label: 'HTML',
+      langIdentifier: 'html',
+    },
+    css: {
+      label: 'CSS',
+      langIdentifier: 'css',
+    },
+    js: {
+      label: 'JS',
+      langIdentifier: 'js',
+    },
+    jinja: {
+      label: 'Jinja',
+      // While `jinja2` is an option on Prism, it does not seem to highlight syntax properly. So use HTML instead.
+      langIdentifier: 'html',
+    },
+  };
+
   // throttling function calls, by Remy Sharp
   // http://remysharp.com/2010/07/21/throttling-function-calls/
   var throttle = function (fn, delay) {
```

Comment:
Agree, it's getting tedious. I tried sharing the non-example usages to a shared utils directory, but I intermittently get this error:

<img width="710" alt="Screenshot 2024-08-29 at 4 30 34 PM" src="https://github.com/user-attachments/assets/58231a71-380b-4f95-807f-c0c7805de595">

Will file this as a separate issue.

Reply:
I suppose if we wanted to do this while still using versioned_static, we could include the utils file in root with versioned_static, and have that file place its functions in global scope, rather than including them into other JS files.

Reply:
Moved to window scope

Reply:
Well, that's one of the reasons why it haven't been done before, and why doing such things as drive-bys may snowball. 

I have nothing agains having throttle moved out from here, but in terms of other component examples we wanted it to explicitly in the script, so it can be copy-pasted. Otherwise, we would provide JS snippet, that doesn't work becuase `throttle` is not implemented.

So overall, I'm not sure what the immediate benefit of extracting this in this particular PR is.

Reply:
@pastelcyborg Yeah that's my understanding and intention as well, but it looks like they got re-extracted by a rebase, I'll fix that now

Reply:
@pastelcyborg Reverted those changes.